modules/check/variables.tf

```
@@ -0,0 +1,58 @@
+# special variables
+
+variable "check_type" {
+  type        = string
+  description = "The type of the check the module should create."
+  validation {
+    condition     = contains(["manual", null], var.check_type)
+    error_message = format("Invalid check type.  Valid values are: %s", join(",", var.check_types))
+  }
+  default = null
+}
+
```

```suggestion
locals {
  check_types = ["manual", "package_version"]
}

variable "type" {
  type        = string
  description = "The type check to create."
  validation {
    condition = contains(check_types, var.type)
    error_message = format("expected type to be one of %v", check_types)
  }
}

```

null is not a valid option - we need this to ALWAYS be specified